test/hotspot/jtreg/serviceability/jvmti/vthread/StopThreadTest/StopThreadTest.java

```diff
@@ -113,6 +113,8 @@ public static void run() {
                 testTaskThread = Thread.ofPlatform().name("TestTaskThread").start(testTask);
             }
             TestTask.ensureBlockedAfterPointA(testTaskThread);
+            /* Sleep for 2 seconds to allow the virtual thread to preempt and unmount. */
+            TestTask.sleep(2000);
```

Comment:
What guarantees the target thread will _ever_ be unmounted? If the test environment has sufficient physical threads, there will be no contention for a carrier thread and thus no motivation to pay the expense of unmounting a virtual thread.

Reply:
It is part of our implementation for JEP 491. We unmount a virtual thread from its carrier thread when it fails to acquire a monitor, allowing the carrier thread to be reused by the scheduler to run other virtual threads. The test case ensures that the monitor remains held indefinitely by another thread, preventing its release. This setup guarantees that the virtual thread will be unmounted by the VM as expected.

Reply:
@fengxue-IS Can you confirm if my above statement is correct? Did I miss any optimizations where we conditionally preempt?

Reply:
this is correct